lib/Authentication/AuthTokens/OIDCAuthToken.php

```diff
@@ -0,0 +1,208 @@
+     * telephone numbers etc) to be stored in a convenient location.
+     * {@see IAuthentication::getDetails()}
+     *
+     * @return Object or null if not used
+     */
+    public function getDetails()
+    {
+        return $this->userDetails;
+    }
+
+    /**
+     * {@see IAuthentication::getPrinciple()}
+     * @return string unique principle string of user
+     */
+    public function getPrinciple()
+    {
+        return $this->principal;
+    }
+
+    /**
+     * {@see IAuthentication::setAuthorities($authorities)}
+     */
+    public function setAuthorities($authorities)
+    {
+        $this->authorities = $authorities;
+    }
+
+    /**
+     * {@see IAuthentication::setDetails($userDetails)}
+     * @param Object $userDetails
+     */
+    public function setDetails($userDetails)
+    {
+        $this->userDetails = $userDetails;
+    }
+
+    /**
+     * {@see IAuthentication::validate()}
+     */
+    public function validate()
+    {
+    }
+
+    /**
+     * {@see IAuthentication::isPreAuthenticating()}
+     */
+    public static function isPreAuthenticating()
+    {
+        return true;
+    }
+
+    /**
+     * Returns true, this token reads the session attributes and so
+     * does not need to be stateful itself.
+     * {@see IAuthentication::isStateless()}
+     */
+    public static function isStateless()
+    {
+        return true;
+    }
+
+    /**
+     * Set principal/User details from the session and check group membership.
+     */
+    protected function setTokenFromSession()
+    {
+        if (in_array($_SERVER['OIDC_CLAIM_iss'], $this->acceptedIssuers, true)) {
+            $this->principal = $_SERVER['REMOTE_USER'];
+            $this->userDetails = array(
+                'AuthenticationRealm' => array($this->authRealm)
+            );
+
+            // Check group membership is acceptable.
+            $this->checkBannedGroups();
+            $this->checkRequiredGroups();
+        }
+    }
+
+    /**
+     * Check the token lists all the required groups.
+     */
+    protected function checkRequiredGroups()
+    {
+        $groupArray = explode(
+            $this->groupSplitChar,
+            $_SERVER[$this->groupHeader]
+        );
+
+        // Build up a list of missing groups.
+        $missingGoodGroups = [];
+        foreach ($this->requiredGroups as $group) {
+            if (!in_array($group, $groupArray)) {
+                $missingGoodGroups[] = $group;
+            }
+        }
+
+        // If the list of missing groups is not empty, reject the user.
+        if (!empty($missingGoodGroups)) {
+            $this->rejectUser(
+                'You are missing the following group(s):',
+                $missingGoodGroups
+            );
+        }
+    }
+
+    /**
+     * Check the token lists none of the banned groups.
+     */
+    protected function checkBannedGroups()
+    {
+        $groupArray = explode($this->groupSplitChar, $_SERVER[$this->groupHeader]);
```

Comment:
this line was split into 3 in the function above. should this be made consistent to it?

```suggestion
        $groupArray = explode(
                $this->groupSplitChar,
                $_SERVER[$this->groupHeader]
        );
```